File: Leetcode/python-solutions/maxaverage.py
class Solution(object):
    def findMaxAverage(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: float
        """
        n = len(nums)
        numSum = sum(nums[:k])
        maxSum = numSum
        i = k
        while i < n:
            numSum += nums[i] - nums[i - k]
            maxSum = max(maxSum, numSum)
            i += 1
        return maxSum / float(k)

File: Leetcode/javasript-problems/maxaverage.js
var findMaxAverage = function(nums, k) {
    let n = nums.length;
    let numSum = 0;

    for (let i = 0; i < k; i++) {
        numSum += nums[i];
    }

    let maxSum = numSum;
    for (let i = k; i < n; i++) {
        numSum += nums[i] - nums[i - k];
        maxSum = Math.max(maxSum, numSum);
    }
    return maxSum / k;
};
